packages/framework/src/validators/zod.validator.ts

```diff
@@ -1,19 +1,45 @@
-import { ZodSchema } from 'zod';
-
-import type { FromSchema, FromSchemaUnvalidated, JsonSchema, Schema } from '../types/schema.types';
+import type {
+  FromSchema,
+  FromSchemaUnvalidated,
+  Schema,
+  JsonSchema,
+  ZodSchemaMinimal,
+  ZodSchema,
+} from '../types/schema.types';
 import type { ValidateResult, Validator } from '../types/validator.types';
+import { checkDependencies } from '../utils/import.utils';
+import { ImportRequirement } from '../types/import.types';
 
 export class ZodValidator implements Validator<ZodSchema> {
-  canHandle(schema: Schema): schema is ZodSchema {
-    return (schema as ZodSchema).safeParseAsync !== undefined;
+  readonly requiredImports: readonly ImportRequirement[] = [
+    {
+      name: 'zod',
+      import: import('zod'),
```

Comment:
Is it better to pass the name in this list and let all the async import handling happen in `checkDependencies`? I see two marginal benefits with that approach:

1. Simpler `ImportRequirement` interface.
2. Handle asynchronicity in one place.

Reply:
Unfortunately this approach doesn't work, see [this comment](https://github.com/novuhq/novu/blob/7e314123d5faaf819f5762387cae793f7b129b18/packages/framework/src/types/import.types.ts#L15:L16) in the code. An explicit `import(...)` statement is needed to ensure the bundle makes the dependency available for import.